CoreScriptsRoot/Modules/Server/ServerChat/ChatChannel.lua

```diff
@@ -57,14 +57,50 @@ function methods:SendMessageObjToFilters(message, messageObj, fromSpeaker)
 	return newMessage
 end
 
-function methods:SendMessageToSpeaker(message, speakerName, fromSpeaker, extraData)
-	local speaker = self.Speakers[speakerName]
-	if (speaker) then
+function ChatSettingsEnabled()
+	local chatPrivacySettingsSuccess, chatPrivacySettingsValue = pcall(function() return UserSettings():IsUserFeatureEnabled("UserChatPrivacySetting") end)
+	local chatPrivacySettingsEnabled = true
+	if chatPrivacySettingsSuccess then
+		chatPrivacySettingsEnabled = chatPrivacySettingsValue
+	end
+	return chatPrivacySettingsEnabled
+end
+
+function methods:CanCommunicateByUserId(userId1, userId2)
+	if ChatSettingsEnabled() == false then
+		return true
+	end
+	if userId1 < 1 or userId2 < 1 then
+		return true
+	end
+	local success, canCommunicate = pcall(function()
+		return Chat:CanUsersChatAsync(userId1, userId2)
+	end)
+	return success and canCommunicate
+end
+
+function methods:CanCommunicate(speakerObj1, speakerObj2)
+	local player1 = speakerObj1:GetPlayer()
+	local player2 = speakerObj2:GetPlayer()
+	if player1 and player2 then
+		self:CanCommunicateByUserId(player1.UserId, player2.UserId)
+	end
+	return true
```

Comment:
If one player is a bot, we will return true. But this is a bug, we should also return self:CanCommunicateByUserId(player1.UserId, player2.UserId)